### PR TITLE
Expose 3D views to python

### PR DESCRIPTION
**Pull request by uclaros**
## Description
The goal of this PR is to allow creating, listing and closing 3d views from python, matching the functionality of their 2d counterparts.

To achieve this some refactoring and code cleanup took place:
- `Qgs3DNavigationWidget` was taken out of `Qgs3DMapCanvas` and moved to `Qgs3DMapCanvasWidget` where it actually belongs
- `Qgs3DMapCanvas` and `Qgs3DMapTool` classes were moved from _app_ to _3d_ library
- `Qgs3DMapCanvas` absorbed the new `Qgs3DWindow` class (or maybe the other way round, but the name `Qgs3DMapCanvas` made more sense and was kept), leaving us with one less stacked widget when creating 3d views. 

`QgisInterface` received three new methods to create, list and close 3d views:
```cpp
virtual QList< Qgs3DMapCanvas * > mapCanvases3D() = 0;
virtual Qgs3DMapCanvas *createNewMapCanvas3D( const QString &name ) = 0;
virtual void closeMapCanvas3D( const QString &name ) = 0;
```

which should allow to create, manipulate and close 3d view in python:
```python
>>> iface.createNewMapCanvas3D('my new 3d view')
<qgis._3d.Qgs3DMapCanvas object at 0x7fa066368d30>
>>> iface.mapCanvases3D()
[<qgis._3d.Qgs3DMapCanvas object at 0x7fa066368d30>]
>>> iface.closeMapCanvas3D('my new 3d view')
>>> iface.mapCanvases3D()
[]
```

`Qgs3DMapCanvas` is now exposed to python allowing access to its `Qgs3DMapSettings`, `QgsCameraController` and `Qgs3DMapScene`

Funded by Natural Resources Canada

### CODE REVIEW

**Comment by nyalldawson** [comment]
@uclaros can you squash the two class renaming commits?

**Comment by nyalldawson** [comment]
Good move!

We should deprecate Qgs3DMapScene::openScenes() too -- that was a short term ~~solution~~ hack to expose some 3d functionality to python

File: src/3d/qgs3dmapscene.h [hunk 1]
```diff
@@ -194,10 +194,10 @@ class _3D_EXPORT Qgs3DMapScene : public QObject
      * Returns a map of 3D map scenes (by name) open in the QGIS application.
      *
      * \note Only available from the QGIS desktop application.
-     *
+     * \deprecated since QGIS 3.36, use QgisAppInterface::mapCanvases3D() instead.
      * \since QGIS 3.30
      */
-    static QMap< QString, Qgs3DMapScene * > openScenes();
+    Q_DECL_DEPRECATED static QMap< QString, Qgs3DMapScene * > openScenes() SIP_DEPRECATED;
```
Comment (wonder-sk) [inline src/3d/qgs3dmapscene.h:200]: @nyalldawson do you want to deprecate + keep this, or is it okay for you if we remove it altogether?

File: src/app/3d/qgs3dmaptoolidentify.cpp [hunk 2]
```diff
@@ -107,16 +109,16 @@ void Qgs3DMapToolIdentify::mouseReleaseEvent( QMouseEvent *event )
     // estimate search radius
     Qgs3DMapScene *scene = mCanvas->scene();
     const double searchRadiusMM = QgsMapTool::searchRadiusMM();
-    const double pixelsPerMM = mCanvas->logicalDpiX() / 25.4;
+    const double pixelsPerMM = mCanvas->screen()->logicalDotsPerInchX(); // mCanvas->logicalDpiX() / 25.4;
```
Comment (wonder-sk) [inline src/app/3d/qgs3dmaptoolidentify.cpp:112]: looks like `/ 25.4` got lost here

**Review by wonder-sk — COMMENTED** [review]
Looking good!

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**